### ISSUE TITLE
Circular Depency Detected

### ISSUE THREAD

**Issue by NikoBiele** [SciML/LinearSolve.jl#563]
**Describe the bug 🐞**

I get a warning of circular depency, related to a lot of the SciML packages. Not sure if this is the right place to post it.
All of the said packages skipped precompilation. Not sure if this is due to error I made myself, but posting here just to notify.

**Expected behavior**

I expect no warning and for the packages to precompile.

**Minimal Reproducible Example 👇**

Without MRE, we would only be able to help you to a limited extent, and attention to the issue would be limited. to know more about MRE refer to [wikipedia](https://en.wikipedia.org/wiki/Minimal_reproducible_example) and [stackoverflow](https://stackoverflow.com/help/minimal-reproducible-example).

```julia
pkg> update
```

**Error & Stacktrace ⚠️**

```julia
(RayTraceHeatTransfer) pkg> update
    Updating registry at `C:\Users\jalok\.julia\registries\General.toml`
  No Changes to `C:\Users\jalok\Dropbox\Git\RayTraceHeatTransfer.jl\Project.toml`
  No Changes to `C:\Users\jalok\Dropbox\Git\RayTraceHeatTransfer.jl\Manifest.toml`
┌ Warning: Circular dependency detected. Precompilation will be skipped for:
│   Base.PkgId(Base.UUID("d236fae5-4411-538c-8e31-a6e3d9e00b46"), "PreallocationTools")
│   Base.PkgId(Base.UUID("385e4588-a1a0-5c1d-98fa-d45bf6f8ecf9"), "LinearSolveKernelAbstractionsExt")
│   Base.PkgId(Base.UUID("1285c0f1-ff9f-5867-b66e-0f359bcf09ba"), "SpecialFunctionsExt")
│   Base.PkgId(Base.UUID("a2df0a61-553a-563b-aed7-0ce21874eb58"), "SciMLOperatorsStaticArraysCoreExt")
│   Base.PkgId(Base.UUID("b00db79b-61e3-50fb-b26f-2d35b2d9e4ed"), "DiffEqBaseChainRulesCoreExt")
│   Base.PkgId(Base.UUID("8913a72c-1f9b-4ce2-8d82-65094dcecaec"), "NonlinearSolve")
│   Base.PkgId(Base.UUID("0d7ed370-da01-4f52-bd93-41d350b8b718"), "StaticArrayInterface")
│   Base.PkgId(Base.UUID("42296aa8-c874-5f57-b5c1-8d6f5ebd5400"), "RecursiveArrayToolsFastBroadcastExt")
│   Base.PkgId(Base.UUID("0bca4576-84f4-4d90-8ffe-ffa030f20462"), "SciMLBase")
│   Base.PkgId(Base.UUID("78064ecb-8a7e-51fd-a998-f0197ddca326"), "DiffEqBaseCUDAExt")
│   Base.PkgId(Base.UUID("63c18a36-062a-441e-b654-da1e3ab1ce7c"), "KernelAbstractions")
│   Base.PkgId(Base.UUID("bdcacae8-1622-11e9-2a5c-532679323890"), "LoopVectorization")
│   Base.PkgId(Base.UUID("d5829a12-d9aa-46ab-831f-fb7c9ab06edf"), "TriangularSolve")
│   Base.PkgId(Base.UUID("73e54eaf-3344-511d-b088-1ac5413eca63"), "RecursiveArrayToolsSparseArraysExt")
│   Base.PkgId(Base.UUID("7792a7ef-975c-4747-a70f-980b88e8d1da"), "StrideArraysCore")
│   Base.PkgId(Base.UUID("f78586ac-ba87-5ec7-b3a5-84bc1b258ee3"), "ArrayInterfaceCUDAExt")
│   Base.PkgId(Base.UUID("476501e8-09a2-5ece-8869-fb82de89a1fa"), "SLEEFPirates")
│   Base.PkgId(Base.UUID("bb5d69b7-63fc-4a16-80bd-7e42200c7bdb"), "MaybeInplace")
│   Base.PkgId(Base.UUID("f517fe37-dbe3-4b94-8317-1923a5111588"), "Polyester")
│   Base.PkgId(Base.UUID("fb6a15b2-703c-40df-9091-08a04967cfa9"), "CloseOpenIntervals")
│   Base.PkgId(Base.UUID("6044458a-139e-583a-bc2f-a730a9b54282"), "DifferentiationInterfaceFiniteDiffExt")
│   Base.PkgId(Base.UUID("d41bc354-129a-5804-8e4c-c37616107c6c"), "NLSolversBase")
│   Base.PkgId(Base.UUID("7034ab61-46d4-4ed7-9d0f-46aef9175898"), "FastBroadcast")
│   Base.PkgId(Base.UUID("04950c4b-5bc4-5740-952d-02d2c1eb583a"), "LinearSolveRecursiveArrayToolsExt")
│   Base.PkgId(Base.UUID("052768ef-5323-5732-b1bb-66c8b64840ba"), "CUDA")
│   Base.PkgId(Base.UUID("699899dd-2948-59d9-a4f6-ce210f3f1f61"), "FiniteDiffSparseArraysExt")
│   Base.PkgId(Base.UUID("3d5dd08c-fd9d-11e8-17fa-ed2836048c2f"), "VectorizationBase")
│   Base.PkgId(Base.UUID("85068d23-b5fb-53f1-8204-05c2aba6942f"), "SparseArraysExt")
│   Base.PkgId(Base.UUID("7cf1493d-886e-405c-b582-53b49fa147b6"), "RayTraceHeatTransfer")
│   Base.PkgId(Base.UUID("9f049cbb-7c7d-5dfe-91f7-cf323d5306ff"), "SparseDiffToolsPolyesterExt")
│   Base.PkgId(Base.UUID("10f19ff3-798f-405d-979b-55457f8fc047"), "LayoutPointers")
│   Base.PkgId(Base.UUID("13011619-4c7c-5ef0-948f-5fc81565cd05"), "AtomixCUDAExt")
│   Base.PkgId(Base.UUID("66d79d19-2cc4-5b0b-ac7a-b340256d1ecd"), "LinearAlgebraExt")
│   Base.PkgId(Base.UUID("aeb06bb4-539b-5a1b-8332-034ed9f8ca66"), "DiffEqBaseUnitfulExt")
│   Base.PkgId(Base.UUID("6425d1ab-ad86-5f6e-82f1-f5118e0dc800"), "EnzymeExt")
│   Base.PkgId(Base.UUID("4676cac9-c8e0-5d6e-a4e0-e3351593cdf5"), "SciMLBaseChainRulesCoreExt")
│   Base.PkgId(Base.UUID("6a86dc24-6348-571c-b903-95158fe2bd41"), "FiniteDiff")
│   Base.PkgId(Base.UUID("499d2952-c3ce-5339-96ee-b197ee6daaa6"), "SpecialFunctionsExt")
│   Base.PkgId(Base.UUID("eae2faf6-b232-58cb-a410-7764fda2830c"), "ChainRulesCoreExt")
│   Base.PkgId(Base.UUID("073a8d7d-86ee-5d75-9348-f9bf6155b014"), "SimpleNonlinearSolveChainRulesCoreExt")
│   Base.PkgId(Base.UUID("47a9eef4-7e08-11e9-0b38-333d64bd3804"), "SparseDiffTools")
│   Base.PkgId(Base.UUID("06b0261c-7a9b-5753-9bdf-fd6840237b4a"), "StaticArrayInterfaceStaticArraysExt")
│   Base.PkgId(Base.UUID("a8fab69c-de6f-5f74-aba3-bef19c2d033f"), "StaticArrayInterfaceOffsetArraysExt")
│   Base.PkgId(Base.UUID("c3ed20f3-746d-5a71-99a7-df1cca8c2b90"), "EnzymeCoreExt")
│   Base.PkgId(Base.UUID("c0aeaf25-5076-4817-a8d5-81caf7dfa961"), "SciMLOperators")
│   Base.PkgId(Base.UUID("731186ca-8d62-57ce-b412-fbd966d074cd"), "RecursiveArrayTools")
│   Base.PkgId(Base.UUID("727e6d20-b764-4bd8-a329-72de5adea6c7"), "SimpleNonlinearSolve")
│   Base.PkgId(Base.UUID("9c97de4a-2903-5b9c-add6-4b1aaffe4ab8"), "MaybeInplaceSparseArraysExt")
│   Base.PkgId(Base.UUID("d3d80556-e9d4-5f37-9878-2ab0fcc64255"), "LineSearches")
│   Base.PkgId(Base.UUID("53ae85a6-f571-4167-b2af-e1d143709226"), "SciMLStructures")
│   Base.PkgId(Base.UUID("7ed4a6bd-45f5-4d41-b270-4a48e9bafcae"), "LinearSolve")
│   Base.PkgId(Base.UUID("2efcf032-c050-4f8e-a9bb-153293bab1f5"), "SymbolicIndexingInterface")
│   Base.PkgId(Base.UUID("14203109-85fb-5f77-af23-1cb7d9032242"), "RecursiveArrayToolsForwardDiffExt")
│   Base.PkgId(Base.UUID("9985400b-97ec-5583-b534-4f70b643bcf7"), "SciMLOperatorsSparseArraysExt")
│   Base.PkgId(Base.UUID("4131c53f-b1d6-5635-a7a3-57f6f930b644"), "DiffEqBaseSparseArraysExt")
│   Base.PkgId(Base.UUID("2b5f629d-d688-5b77-993f-72d75c75574e"), "DiffEqBase")
│   Base.PkgId(Base.UUID("e24d4dde-ed20-5ee7-b465-f1dd73f9b6ba"), "LinearSolveCUDAExt")
│   Base.PkgId(Base.UUID("4b8ca102-9d0f-5df8-84ed-253e0f005ba7"), "ForwardDiffExt")
│   Base.PkgId(Base.UUID("f2c3362d-daeb-58d1-803e-2bc74f2840b4"), "RecursiveFactorization")
│   Base.PkgId(Base.UUID("75e56524-3a34-51de-85ea-03aa6eac4b64"), "FiniteDiffStaticArraysExt")
└ @ Base.Precompilation precompilation.jl:511

(RayTraceHeatTransfer) pkg> 
```

**Environment (please complete the following information):**

  - Output of `using Pkg; Pkg.status()`

```julia
julia> Pkg.status()
Project RayTraceHeatTransfer v0.2.1
Status `C:\Users\jalok\Dropbox\Git\RayTraceHeatTransfer.jl\Project.toml`
  [6e4b80f9] BenchmarkTools v1.5.0
  [336ed68f] CSV v0.10.15
⌃ [052768ef] CUDA v5.5.0
⌅ [a93c6f00] DataFrames v1.6.1
  [864edb3b] DataStructures v0.18.20
  [a98d9a8b] Interpolations v0.15.1
  [b964fa9f] LaTeXStrings v1.4.0
⌅ [7ed4a6bd] LinearSolve v2.30.2
⌅ [8913a72c] NonlinearSolve v3.13.1
  [91a5bcdd] Plots v1.40.9
  [0a514795] SparseMatrixColorings v0.4.10
  [90137ffa] StaticArrays v1.9.8
  [37e2e46d] LinearAlgebra v1.11.0
  [8dfed614] Test v1.11.0
Info Packages marked with ⌃ and ⌅ have new versions available. Those with ⌃ may be upgradable, but those with ⌅ are restricted by compatibility constraints from upgrading. To see why use `status --outdated`

julia> 
```

  - Output of `using Pkg; Pkg.status(; mode = PKGMODE_MANIFEST)`

```julia
julia> Pkg.status(; mode = PKGMODE_MANIFEST)
Project RayTraceHeatTransfer v0.2.1
Status `C:\Users\jalok\Dropbox\Git\RayTraceHeatTransfer.jl\Manifest.toml`
  [47edcb42] ADTypes v1.11.0
  [621f4979] AbstractFFTs v1.5.0
  [7d9f7c33] Accessors v0.1.39
  [79e6a3ab] Adapt v4.1.1
  [66dad0bd] AliasTables v1.1.3
  [ec485272] ArnoldiMethod v0.4.0
  [4fba245c] ArrayInterface v7.18.0
  [4c555306] ArrayLayouts v1.11.0
  [a9b6321e] Atomix v1.0.1
  [13072b0f] AxisAlgorithms v1.1.0
  [ab4f0b2a] BFloat16s v0.5.0
  [6e4b80f9] BenchmarkTools v1.5.0
  [d1d4a3ce] BitFlags v0.1.9
  [62783981] BitTwiddlingConvenienceFunctions v0.1.6
  [fa961155] CEnum v0.5.0
  [2a0fbf3d] CPUSummary v0.2.6
  [336ed68f] CSV v0.10.15
⌃ [052768ef] CUDA v5.5.0
  [1af6417a] CUDA_Runtime_Discovery v0.3.5
  [d360d2e6] ChainRulesCore v1.25.0
  [fb6a15b2] CloseOpenIntervals v0.1.13
  [944b1d66] CodecZlib v0.7.6
  [35d6a980] ColorSchemes v3.27.1
  [3da002f7] ColorTypes v0.12.0
  [c3611d14] ColorVectorSpace v0.11.0
  [5ae59095] Colors v0.13.0
  [38540f10] CommonSolve v0.2.4
  [bbf7d656] CommonSubexpressions v0.3.1
  [f70d9fcc] CommonWorldInvalidations v1.0.0
  [34da2185] Compat v4.16.0
  [a33af91c] CompositionsBase v0.1.2
  [2569d6c7] ConcreteStructs v0.2.3
  [f0e56b4a] ConcurrentUtilities v2.4.3
  [187b0558] ConstructionBase v1.5.8
  [d38c429a] Contour v0.6.3
  [adafc99b] CpuId v0.3.1
  [a8cc5b0e] Crayons v4.1.1
  [9a962f9c] DataAPI v1.16.0
⌅ [a93c6f00] DataFrames v1.6.1
  [864edb3b] DataStructures v0.18.20
  [e2d170a0] DataValueInterfaces v1.0.0
  [8bb1440f] DelimitedFiles v1.9.1
  [2b5f629d] DiffEqBase v6.160.0
  [163ba53b] DiffResults v1.1.0
  [b552c78f] DiffRules v1.15.1
⌃ [a0c0ee7d] DifferentiationInterface v0.6.26
  [ffbed154] DocStringExtensions v0.9.3
  [4e289a0a] EnumX v1.0.4
⌃ [f151be2c] EnzymeCore v0.7.8
  [460bff9d] ExceptionUnwrapping v0.1.11
  [e2ba6199] ExprTools v0.1.10
⌅ [6b7a57c9] Expronicon v0.8.5
  [c87230d0] FFMPEG v0.4.2
  [7034ab61] FastBroadcast v0.3.5
  [9aa1b823] FastClosures v0.3.2
  [29a986be] FastLapackInterface v2.0.4
  [a4df4552] FastPower v1.1.1
  [48062228] FilePathsBase v0.9.22
  [1a297f60] FillArrays v1.13.0
  [6a86dc24] FiniteDiff v2.26.2
  [53c48c17] FixedPointNumbers v0.8.5
  [1fa38f19] Format v1.3.7
  [f6369f11] ForwardDiff v0.10.38
  [069b7b12] FunctionWrappers v1.1.3
  [77dc65aa] FunctionWrappersWrappers v0.1.3
⌅ [0c68f7d7] GPUArrays v10.3.1
⌅ [46192b85] GPUArraysCore v0.1.6
⌅ [61eb1bfa] GPUCompiler v0.27.8
  [28b8d3ca] GR v0.73.9
  [86223c79] Graphs v1.12.0
  [42e2da0e] Grisu v1.0.2
  [cd3eb016] HTTP v1.10.14
  [3e5b6fbb] HostCPUFeatures v0.1.17
  [615f187c] IfElse v0.1.1
  [d25df0c9] Inflate v0.1.5
  [842dd82b] InlineStrings v1.4.2
  [a98d9a8b] Interpolations v0.15.1
  [3587e190] InverseFunctions v0.1.17
  [41ab1584] InvertedIndices v1.3.1
  [92d709cd] IrrationalConstants v0.2.2
  [82899510] IteratorInterfaceExtensions v1.0.0
  [1019f520] JLFzf v0.1.9
  [692b3bcd] JLLWrappers v1.6.1
  [682c06a0] JSON v0.21.4
  [ef3ab10e] KLU v0.6.0
  [63c18a36] KernelAbstractions v0.9.31
  [ba0b0d4f] Krylov v0.9.8
  [929cbde3] LLVM v9.1.3
  [8b046642] LLVMLoopInfo v1.0.0
  [b964fa9f] LaTeXStrings v1.4.0
  [23fbe1c1] Latexify v0.16.5
  [10f19ff3] LayoutPointers v0.1.17
  [5078a376] LazyArrays v2.3.1
  [d3d80556] LineSearches v7.3.0
⌅ [7ed4a6bd] LinearSolve v2.30.2
  [2ab3a3ac] LogExpFunctions v0.3.29
  [e6f89c97] LoggingExtras v1.1.0
  [bdcacae8] LoopVectorization v0.12.171
  [d8e11817] MLStyle v0.4.17
  [1914dd2f] MacroTools v0.5.13
  [d125e4d3] ManualMemory v0.1.8
  [bb5d69b7] MaybeInplace v0.1.4
  [739be429] MbedTLS v1.1.9
  [442fdcdd] Measures v0.3.2
  [e1d29d7a] Missings v1.2.0
  [46d2c3a1] MuladdMacro v0.2.4
  [d41bc354] NLSolversBase v7.8.3
  [5da4648a] NVTX v0.3.5
  [77ba4419] NaNMath v1.0.2
⌅ [8913a72c] NonlinearSolve v3.13.1
  [6fe1bfb0] OffsetArrays v1.14.2
  [4d8831e6] OpenSSL v1.4.3
  [bac558e1] OrderedCollections v1.7.0
  [65ce6f38] PackageExtensionCompat v1.0.2
  [d96e819e] Parameters v0.12.3
  [69de0a69] Parsers v2.8.1
  [b98c9c47] Pipe v1.3.0
  [ccf2f8ad] PlotThemes v3.3.0
  [995b91a9] PlotUtils v1.4.3
  [91a5bcdd] Plots v1.40.9
  [f517fe37] Polyester v0.7.16
  [1d0040c9] PolyesterWeave v0.2.2
  [2dfb63ee] PooledArrays v1.4.3
  [d236fae5] PreallocationTools v0.4.24
  [aea7be01] PrecompileTools v1.2.1
  [21216c6a] Preferences v1.4.3
  [08abe8d2] PrettyTables v2.4.0
  [43287f4e] PtrArrays v1.2.1
  [74087812] Random123 v1.7.0
  [e6cf234a] RandomNumbers v1.6.0
  [c84ed2f1] Ratios v0.4.5
  [3cdcf5f2] RecipesBase v1.3.4
  [01d81517] RecipesPipeline v0.6.12
  [731186ca] RecursiveArrayTools v3.27.4
  [f2c3362d] RecursiveFactorization v0.2.23
  [189a3867] Reexport v1.2.2
  [05181044] RelocatableFolders v1.0.1
  [ae029012] Requires v1.3.0
  [7e49a35a] RuntimeGeneratedFunctions v0.5.13
  [94e857df] SIMDTypes v0.1.0
  [476501e8] SLEEFPirates v0.6.43
  [0bca4576] SciMLBase v2.68.1
  [c0aeaf25] SciMLOperators v0.3.12
  [53ae85a6] SciMLStructures v1.6.1
  [6c6a2e73] Scratch v1.2.1
  [91c51154] SentinelArrays v1.4.8
  [efcf1570] Setfield v1.1.1
  [992d4aef] Showoff v1.0.3
  [777ac1f9] SimpleBufferStream v1.2.0
⌅ [727e6d20] SimpleNonlinearSolve v1.12.3
  [699a6c99] SimpleTraits v0.9.4
  [a2af1166] SortingAlgorithms v1.2.1
  [47a9eef4] SparseDiffTools v2.23.0
  [0a514795] SparseMatrixColorings v0.4.10
  [e56a9233] Sparspak v0.3.9
  [276daf66] SpecialFunctions v2.5.0
  [860ef19b] StableRNGs v1.0.2
  [aedffcd0] Static v1.1.1
  [0d7ed370] StaticArrayInterface v1.8.0
  [90137ffa] StaticArrays v1.9.8
  [1e83bf80] StaticArraysCore v1.4.3
  [10745b16] Statistics v1.11.1
  [82ae8749] StatsAPI v1.7.0
  [2913bbd2] StatsBase v0.34.4
  [7792a7ef] StrideArraysCore v0.5.7
  [892a3eda] StringManipulation v0.4.0
  [2efcf032] SymbolicIndexingInterface v0.3.36
  [3783bdb8] TableTraits v1.0.1
  [bd369af6] Tables v1.12.0
  [62fd8b95] TensorCore v0.1.1
  [8290d209] ThreadingUtilities v0.5.2
  [a759f4b9] TimerOutputs v0.5.26
  [3bb67fe8] TranscodingStreams v0.11.3
  [d5829a12] TriangularSolve v0.2.1
  [781d530d] TruncatedStacktraces v1.4.0
  [5c2747f8] URIs v1.5.1
  [3a884ed6] UnPack v1.0.2
  [1cfade01] UnicodeFun v0.4.1
  [1986cc42] Unitful v1.21.1
  [45397f5d] UnitfulLatexify v1.6.4
  [013be700] UnsafeAtomics v0.3.0
  [41fe7b60] Unzip v0.2.0
  [3d5dd08c] VectorizationBase v0.21.71
  [19fa3120] VertexSafeGraphs v0.2.0
  [ea10d353] WeakRefStrings v1.4.2
  [efce3f68] WoodburyMatrices v1.0.0
  [76eceee3] WorkerUtilities v1.6.1
  [6e34b625] Bzip2_jll v1.0.8+2
⌅ [4ee394cb] CUDA_Driver_jll v0.10.4+0
  [76a88914] CUDA_Runtime_jll v0.15.5+0
  [83423d85] Cairo_jll v1.18.2+1
  [ee1fde0b] Dbus_jll v1.14.10+0
  [2702e6a9] EpollShim_jll v0.0.20230411+1
  [2e619515] Expat_jll v2.6.4+1
⌅ [b22a6f82] FFMPEG_jll v4.4.4+1
  [a3f928ae] Fontconfig_jll v2.15.0+0
  [d7e528f0] FreeType2_jll v2.13.3+1
  [559328eb] FriBidi_jll v1.0.14+0
  [0656b61e] GLFW_jll v3.4.0+1
  [d2c73de3] GR_jll v0.73.9+0
  [78b55507] Gettext_jll v0.21.0+0
  [7746bdde] Glib_jll v2.82.2+1
  [3b182d85] Graphite2_jll v1.3.14+1
  [2e76f6c2] HarfBuzz_jll v8.5.0+0
  [1d5cc7b8] IntelOpenMP_jll v2024.2.1+0
  [aacddb02] JpegTurbo_jll v3.0.4+0
  [9c1d0b0a] JuliaNVTXCallbacks_jll v0.2.1+0
  [c1c5ebd0] LAME_jll v3.100.2+0
  [88015f11] LERC_jll v4.0.0+0
  [dad2f222] LLVMExtra_jll v0.0.34+0
  [1d63c593] LLVMOpenMP_jll v18.1.7+0
  [dd4b983a] LZO_jll v2.10.2+1
⌅ [e9f186c6] Libffi_jll v3.2.2+1
  [d4300ac3] Libgcrypt_jll v1.11.0+0
  [7e76a0d4] Libglvnd_jll v1.7.0+0
  [7add5ba3] Libgpg_error_jll v1.50.0+0
  [94ce4f54] Libiconv_jll v1.17.0+1
  [4b2f31a3] Libmount_jll v2.40.2+0
  [89763e89] Libtiff_jll v4.7.0+0
  [38a345b3] Libuuid_jll v2.40.2+0
  [856f044c] MKL_jll v2024.2.0+0
  [e98f9f5b] NVTX_jll v3.1.0+2
  [e7412a2a] Ogg_jll v1.3.5+1
  [458c3c95] OpenSSL_jll v3.0.15+1
  [efe28fd5] OpenSpecFun_jll v0.5.5+0
  [91d4177d] Opus_jll v1.3.3+0
  [36c8627f] Pango_jll v1.54.1+0
⌅ [30392449] Pixman_jll v0.43.4+0
  [c0090381] Qt6Base_jll v6.7.1+1
  [629bc702] Qt6Declarative_jll v6.7.1+2
  [ce943373] Qt6ShaderTools_jll v6.7.1+1
  [e99dba38] Qt6Wayland_jll v6.7.1+1
  [a44049a8] Vulkan_Loader_jll v1.3.243+0
  [a2964d1f] Wayland_jll v1.21.0+1
  [2381bf8a] Wayland_protocols_jll v1.31.0+0
  [02c8fc9c] XML2_jll v2.13.5+0
  [aed1982a] XSLT_jll v1.1.42+0
  [ffd25f8a] XZ_jll v5.6.3+0
  [f67eecfb] Xorg_libICE_jll v1.1.1+0
  [c834827a] Xorg_libSM_jll v1.2.4+0
  [4f6342f7] Xorg_libX11_jll v1.8.6+1
  [0c0b7dd1] Xorg_libXau_jll v1.0.11+1
  [935fb764] Xorg_libXcursor_jll v1.2.0+4
  [a3789734] Xorg_libXdmcp_jll v1.1.4+1
  [1082639a] Xorg_libXext_jll v1.3.6+1
  [d091e8ba] Xorg_libXfixes_jll v5.0.3+4
  [a51aa0fd] Xorg_libXi_jll v1.7.10+4
  [d1454406] Xorg_libXinerama_jll v1.1.5+0
  [ec84b674] Xorg_libXrandr_jll v1.5.4+0
  [ea2f1a96] Xorg_libXrender_jll v0.9.11+1
  [14d82f49] Xorg_libpthread_stubs_jll v0.1.1+1
  [c7cfdc94] Xorg_libxcb_jll v1.17.0+1
  [cc61e674] Xorg_libxkbfile_jll v1.1.2+1
  [e920d4aa] Xorg_xcb_util_cursor_jll v0.1.4+0
  [12413925] Xorg_xcb_util_image_jll v0.4.0+1
  [2def613f] Xorg_xcb_util_jll v0.4.0+1
  [975044d2] Xorg_xcb_util_keysyms_jll v0.4.0+1
  [0d47668e] Xorg_xcb_util_renderutil_jll v0.3.9+1
  [c22f9ab0] Xorg_xcb_util_wm_jll v0.4.1+1
  [35661453] Xorg_xkbcomp_jll v1.4.6+1
  [33bec58e] Xorg_xkeyboard_config_jll v2.39.0+0
  [c5fb5394] Xorg_xtrans_jll v1.5.0+1
  [3161d3a3] Zstd_jll v1.5.6+1
  [35ca27e7] eudev_jll v3.2.9+0
  [214eeab7] fzf_jll v0.56.3+0
  [1a1c6b14] gperf_jll v3.1.1+1
  [a4ae2306] libaom_jll v3.9.0+0
  [0ac62f75] libass_jll v0.15.2+0
  [1183f4f0] libdecor_jll v0.2.2+0
  [2db6ffa8] libevdev_jll v1.11.0+0
  [f638f0a6] libfdk_aac_jll v2.0.3+0
  [36db933b] libinput_jll v1.18.0+0
  [b53b4c65] libpng_jll v1.6.44+0
  [f27f6e37] libvorbis_jll v1.3.7+2
  [009596ad] mtdev_jll v1.1.6+0
  [1317d2d5] oneTBB_jll v2021.12.0+0
⌅ [1270edf5] x264_jll v2021.5.5+0
⌅ [dfaa095f] x265_jll v3.5.0+0
  [d8fb68d0] xkbcommon_jll v1.4.1+1
  [0dad84c5] ArgTools v1.1.2
  [56f22d72] Artifacts v1.11.0
  [2a0f44e3] Base64 v1.11.0
  [ade2ca70] Dates v1.11.0
  [8ba89e20] Distributed v1.11.0
  [f43a241f] Downloads v1.6.0
  [7b1f6079] FileWatching v1.11.0
  [9fa8497b] Future v1.11.0
  [b77e0a4c] InteractiveUtils v1.11.0
  [4af54fe1] LazyArtifacts v1.11.0
  [b27032c2] LibCURL v0.6.4
  [76f85450] LibGit2 v1.11.0
  [8f399da3] Libdl v1.11.0
  [37e2e46d] LinearAlgebra v1.11.0
  [56ddb016] Logging v1.11.0
  [d6f4376e] Markdown v1.11.0
  [a63ad114] Mmap v1.11.0
  [ca575930] NetworkOptions v1.2.0
  [44cfe95a] Pkg v1.11.0
  [de0858da] Printf v1.11.0
  [9abbd945] Profile v1.11.0
  [3fa0cd96] REPL v1.11.0
  [9a3f8284] Random v1.11.0
  [ea8e919c] SHA v0.7.0
  [9e88b42a] Serialization v1.11.0
  [1a1011a3] SharedArrays v1.11.0
  [6462fe0b] Sockets v1.11.0
  [2f01184e] SparseArrays v1.11.0
  [f489334b] StyledStrings v1.11.0
  [fa267f1f] TOML v1.0.3
  [a4e569a6] Tar v1.10.0
  [8dfed614] Test v1.11.0
  [cf7118a7] UUIDs v1.11.0
  [4ec0a83e] Unicode v1.11.0
  [e66e0078] CompilerSupportLibraries_jll v1.1.1+0
  [deac9b47] LibCURL_jll v8.6.0+0
  [e37daf67] LibGit2_jll v1.7.2+0
  [29816b5a] LibSSH2_jll v1.11.0+1
  [c8ffd9c3] MbedTLS_jll v2.28.6+0
  [14a3606d] MozillaCACerts_jll v2023.12.12
  [4536629a] OpenBLAS_jll v0.3.27+1
  [05823500] OpenLibm_jll v0.8.1+2
  [efcefdf7] PCRE2_jll v10.42.0+1
  [bea87d4a] SuiteSparse_jll v7.7.0+0
  [83775a58] Zlib_jll v1.2.13+1
  [8e850b90] libblastrampoline_jll v5.11.0+0
  [8e850ede] nghttp2_jll v1.59.0+0
  [3f19e933] p7zip_jll v17.4.0+2
Info Packages marked with ⌃ and ⌅ have new versions available. Those with ⌃ may be upgradable, but those with ⌅ are restricted by compatibility constraints from upgrading. To see why use `status --outdated -m`

julia> 
```

  - Output of `versioninfo()`

```julia
julia> versioninfo()
Julia Version 1.11.1
Commit 8f5b7ca12a (2024-10-16 10:53 UTC)
Build Info:
  Official https://julialang.org/ release
Platform Info:
  OS: Windows (x86_64-w64-mingw32)
  CPU: 16 × AMD Ryzen 7 5825U with Radeon Graphics
  WORD_SIZE: 64
  LLVM: libLLVM-16.0.6 (ORCJIT, znver3)
Threads: 16 default, 0 interactive, 8 GC (on 16 virtual cores)
Environment:
  JULIA_EDITOR = code
  JULIA_NUM_THREADS = 16

julia> 
```

**Additional context**

**Comment by ChrisRackauckas** [comment]
@tpolarity you had a tool that can help with this?

**Comment by topolarity** [comment]
The tool is Julia 1.11.2 🙂

This cycle is probably erroneous, caused by a bug we fixed in 1.11.2 (https://github.com/JuliaLang/julia/pull/55910#issuecomment-2488843885)

If not, the error message is also much better in that release and should tell you the actual cycle that was detected

**Comment by NikoBiele** [comment]
Thanks for your replies,  I will update my Julia version, should solve it 😊
